lib/redcord/prepared_redis.rb

```diff
@@ -12,11 +12,16 @@ class Redcord::PreparedRedis < Redis
     ).returns(Integer)
   end
   def create_hash_returning_id(key, args)
-    evalsha(
-      self.class.server_script_shas[:create_hash_returning_id],
-      keys: [key],
-      argv: args.to_a.flatten,
-    ).to_i
+    Redcord::Base.trace(
```

Comment:
I wonder if the public API methods could be auto instrumented instead of manually?

Reply:
there's auto instrumentation for redis (if the supplied datadog.tracer is configured to do so).

or do you mean to do some runtime method generations to define those methods?

Reply:
> or do you mean to do some runtime method generations to define those methods?

Yea I guess I wasn't sure if there was already some automation in place. If there is, the tracing could be part of generating / instrumenting that code instead of manually.